pkg/storegateway/indexheader/encoding/reader.go

```diff
@@ -138,10 +138,3 @@ func (f *FileReader) Size() int {
 func (f *FileReader) Len() int {
 	return f.length - f.pos
 }
-
-// close closes the underlying resources used by this FileReader. This method
```

Comment:
Should we add a note to `FileReader` that it is intended to be used in conjunction with a `DecbufFactory`, and that if it isn't, it'll leak file handles?

Reply:
How about unexporting completely the `FileReader` (into `fileReader`)? It's not intended to be used from outside, and can't be used properly from outside of this package. It's just a file abstraction used by `DecBuf` and doesn't need to be exposed.